rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/attachment/Sw360AttachmentService.java

```diff
@@ -569,4 +570,71 @@ public void fillCheckedAttachmentData(Attachment attachment, User user) throws S
             attachment.unsetCheckedOn();
         }
     }
+    public boolean isValidSbomFile(MultipartFile file) throws  TException {
+        if (file == null) {
+            return false;
+        }
+
+        byte[] buffer = new byte[1024];
+        try (InputStream inputStream = file.getInputStream()) {
+            int bytesRead = inputStream.read(buffer);
+            if (bytesRead < 1) return false;
+
+            String contentStart = new String(buffer, 0, bytesRead, StandardCharsets.UTF_8).trim();
+
+            // SPDX JSON format
+            if (contentStart.startsWith("{") && contentStart.contains("\"spdxVersion\"")) {
+                return true;
+            }
+
+            // SPDX RDF/XML format
+            if (contentStart.startsWith("<?xml")) {
+                if (contentStart.contains("xmlns:spdx=\"http://spdx.org/rdf/terms#\"") ||
+                        contentStart.contains("xmlns=\"http://spdx.org/rdf/terms#\"")) {
+                    return true;
+                }
+
+                if (contentStart.contains("<spdx:SpdxDocument") ||
+                        contentStart.contains("spdx:name") ||
+                        contentStart.contains("SPDXRef-")) {
+                    return true;
+                }
+
+                if (contentStart.contains("SPDX-")) {
+                    return true;
+                }
+            }
+
+            // CycloneDX JSON format
+            if (contentStart.startsWith("{") &&
+                    (contentStart.contains("\"bomFormat\": \"CycloneDX\"") ||
+                            contentStart.contains("\"bomFormat\":\"CycloneDX\""))) {
+                return true;
+            }
+
+            if (contentStart.startsWith("<?xml")) {
+                // Check for standard CycloneDX attributes and namespaces
+                if (contentStart.contains("bomFormat=\"CycloneDX\"") ||
+                        contentStart.contains("xmlns=\"http://cyclonedx.org/schema/bom\"") ||
+                        contentStart.contains("xmlns:bom=\"http://cyclonedx.org/schema/bom\"") ||
+                        contentStart.contains("http://cyclonedx.org/schema/bom/")) {
+                    return true;
+                }
+
+                if (contentStart.contains("<cycloneDx:") ||
+                        contentStart.contains("<bom:") ||
+                        contentStart.contains("<cyclonedx") ||
+                        contentStart.contains("CycloneDX format")) {
+                    return true;
+                }
+            }
```

Comment:
Same can be done with [‎Parser::validate‎](https://github.com/CycloneDX/cyclonedx-core-java/blob/b0bdc500ddda904598d8439bd57ea07312060497/src/main/java/org/cyclonedx/parsers/Parser.java#L75) library function.